interactions/api/gateway/client.py

```diff
@@ -311,13 +311,47 @@ def _dispatch_event(self, event: str, data: dict) -> None:  # sourcery no-metric
 
                     if _context.data._json.get("options"):
                         for option in _context.data.options:
-
-                            if option.focused:
-                                __name, _value = self.__sub_command_context(option, _context)
-                                _name += f"_{__name}" if __name else ""
-
-                                if _value:
-                                    __args.append(_value)
+                            if isinstance(option, dict):
+                                option = Option(**option)
+                            if option.type not in (
+                                OptionType.SUB_COMMAND,
+                                OptionType.SUB_COMMAND_GROUP,
+                            ):
+                                if option.focused:
+                                    __name, _value = self.__sub_command_context(option, _context)
+                                    _name += f"_{__name}" if __name else ""
+                                    if _value:
+                                        __args.append(_value)
+
+                            elif option.type == OptionType.SUB_COMMAND:
+                                for _option in option.options:
+                                    if isinstance(_option, dict):
+                                        _option = Option(**_option)
+                                    if _option.focused:
+                                        __name, _value = self.__sub_command_context(
+                                            _option, _context
+                                        )
+                                        _name += f"_{__name}" if __name else ""
+                                        if _value:
+                                            __args.append(_value)
+                                    break
+
+                            elif option.type == OptionType.SUB_COMMAND_GROUP:
+                                for _option in option.options:
+                                    if isinstance(_option, dict):
+                                        _option = Option(**_option)
+                                    for __option in _option.options:
+                                        if isinstance(__option, dict):
+                                            __option = Option(**__option)
+                                        if __option.focused:
+                                            __name, _value = self.__sub_command_context(
+                                                __option, _context
+                                            )
+                                            _name += f"_{__name}" if __name else ""
+                                            if _value:
+                                                __args.append(_value)
+                                        break
+                                    break
```

Comment:
I don't understand how this part works.
Why does it `break` in the first loop?
It only checks if the option is focused for the *first* option if it exists.
Are you not allowed to have autocomplete options past the first option of the command?

Reply:
I think it only parses one option since you can only enter one, anyways, so only one is present. But it was in the code before, so I kept it, whoever made it probably made it with a reason xD